src/Mod/Assembly/App/AssemblyLink.cpp

```diff
@@ -229,11 +241,23 @@ void AssemblyLink::synchronizeComponents()
             if (obj->isDerivedFrom<AssemblyLink>()) {
                 auto* asmLink = static_cast<AssemblyLink*>(obj);
                 auto* subAsmLink = new AssemblyLink();
+                auto* origin = new App::Origin();
+
+
                 doc->addObject(subAsmLink, obj->getNameInDocument());
+                subAsmLink->Origin.setValue(asmLink->Origin.getValue());
                 subAsmLink->LinkedObject.setValue(obj);
                 subAsmLink->Rigid.setValue(asmLink->Rigid.getValue());
                 subAsmLink->Label.setValue(obj->Label.getValue());
+
+                origin->Label.setValue("Origin");
+                auto* test = dynamic_cast<App::PropertyPlacement*>(
+                    subAsmLink->getPropertyByName("Placement"));
+                origin->Placement.setValue(test->getValue());
```

Comment:
Memory leak. You create a new App::Origin object but this object is never deleted. Nor is it added to the document at any point.

252 to 256 is going nothing because the origin object is lost. Besides those lines don't really make sense.